internal/bucket/bucket.go

```
@@ -81,6 +86,38 @@ func validateRelativePath(name, value string) error {
 	return nil
 }
 
+func ValidateBlobHash(hash string) error {
+	if len(hash) != blobHashLength {
+		return fmt.Errorf("invalid hash: must be %d characters", blobHashLength)
+	}
+	for _, r := range hash {
+		if !((r >= 'A' && r <= 'Z') || (r >= 'a' && r <= 'z') || (r >= '0' && r <= '9') || r == '-' || r == '_') {
```

golangci-lint (2.12.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)